.github/workflows/ci.yml

```diff
@@ -24,8 +24,23 @@ jobs:
         toolchain: nightly-2022-12-11
         override: true
         components: rustfmt
+    # Binary compatibility settings from bytecodealliance/wasmtime
+    - name: Binary compatibility settings (macOS)
+      if: ${{ startsWith(matrix.os, 'macos') }}
+      run: echo 'MACOSX_DEPLOYMENT_TARGET=10.9' >> $GITHUB_ENV
+    - name: Binary compatibility settings (Windows)
+      if: ${{ startsWith(matrix.os, 'windows') }}
+      run: echo 'RUSTFLAGS=-Ctarget-feature=+crt-static' >> $GITHUB_ENV
+    # Run gradle normally to build the JNI lib
     - name: Build with Gradle
+      if: ${{ !startsWith(matrix.os, 'ubuntu') }}
       run: ./gradlew build copyJniLib
+    # Run gradle in a docker container to build the JNI lib on Linux for old glibc compatibility
+    - name: Build with Gradle (Linux)
+      if: ${{ startsWith(matrix.os, 'ubuntu') }}
+      run: |
+        docker build -t build-image ./ci/docker/x86_64-linux
+        docker run --rm --volume $PWD:/build --workdir /build build-image ./gradlew build copyJniLib -x javadoc
```

Comment:
I excluded the javadoc gradle task here because it was failing on Java 11 with no output, and I didn't think it was worth debugging since it's not needed here.